core/domain/exp_domain.py

```diff
@@ -1667,6 +1797,20 @@ def validate(self, strict=False):
                             'does not exist in this exploration'
                             % param_change.name)
 
+            for hint in interaction.hints:
```

Comment:
The checks at this level (i.e. the Exploration level) should only be for stuff which needs the whole exploration context for validation (e.g. fallbacks need a destination that points to another state within the exploration). Otherwise, such checks should be moved to the lowest level possible. It looks like all these checks can be scoped to Interaction level (or lower).